playwright/tests/utils.py

```diff
@@ -0,0 +1,111 @@
+"""
+Useful functions for testing.
+"""
+
+from datetime import datetime
+from pathlib import Path
+
+
+def get_hugo_species_dir() -> Path:
+    """
+    Get the Hugo species directory.
+    """
+    return Path(__file__).parent.parent.parent / "hugo/content/species"
+
+
+def get_list_of_species() -> list[str]:
+    """
+    Search the Hugo content directory to get all the species on the website.
+    This helper function is useful for testing each species page in the same way.
+
+    The function also validates each page is not in draft mode (wont show up on website yet)
+    """
+    HUGO_SPECIES_DIR = get_hugo_species_dir()
+
+    species = []
+    for folder in HUGO_SPECIES_DIR.iterdir():
+        if not folder.is_dir():
+            continue
+
+        index_file = folder / "_index.md"
+        if not index_file.exists():
+            continue
+
+        with index_file.open() as file_in:
```

Comment:
So here comes a nitpick, but it is a nice thing to ponder about. This code checking for draft status is difficult to test because: 
1. It is nested in a loop
2. It performs IO

Here is a refactoring suggestion:
```python
# Isolate the logic in a function that takes a string as input, because it is easy to test
def is_draft(content):
  return "draft: true" in content

def list_species():
  return [ folder.name for folder in HUGO_SPECIES_DIR.iterdir()  if (index := folder / "_index.md").exists() and is_draft(index.read_text())]
```
- `read_text()` is nice because it handles opening and closing the file
- the walrus `:=` and list comprehension are just ~to show off~ for the curiosity